flocker/node/agents/test/test_blockdevice.py

```diff
@@ -2310,22 +2314,29 @@ def test_profile_respected(self):
         mandatory profiles.
         """
         for profile in (c.value for c in MandatoryProfiles.iterconstants()):
-            self.api.create_volume_with_profile(profile)
+            self.api.create_volume_with_profile(dataset_id=uuid4(),
```

Comment:
That seems fine.

Unrelatedly, there are a couple things that ideally would be different about this test, I think.

There are some drawbacks to having loops in tests.  If one the call fails for one of the values, the test won't exercise the rest of the values.  So you can't fix the failing case and be confident the test will pass on the next run.  Also, typically failures are such that you can't tell which iteration of the loop the test failed on so you don't actually have a great idea of what the problem is.

Also, this particular test doesn't make any assertions so, in xUnit lingo, it can't fail, it can only "error".  This isn't a huge deal but it's usually not something to aim for: failures are more likely to convey something meaningful to a developer than errors are.

Also, the test is satisfied by a no-op implementation, so not a ton of value there. :smile: